cloudwatch_logs_common/test/cloudwatch_facade_test.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/core/Aws.h>
+#include <aws/core/client/AWSClient.h>
+#include <aws/core/NoResult.h>
+#include <cloudwatch_logs_common/cloudwatch_logs_client_mock.h>
+#include <cloudwatch_logs_common/ros_cloudwatch_logs_errors.h>
+#include <cloudwatch_logs_common/utils/cloudwatch_facade.h>
+#include <gtest/gtest.h>
+
+
+using namespace Aws::CloudWatchLogs::Utils;
+
+const std::string LOG_GROUP_NAME1 = "TestGroup1";
+const std::string LOG_GROUP_NAME2 = "TestGroup2";
+const std::string LOG_STREAM_NAME1 = "TestStream1";
+const std::string LOG_STREAM_NAME2 = "TestStream2";
+
+
+class TestCloudWatchFacade : public ::testing::Test
+{
+protected:
+  std::list<Aws::CloudWatchLogs::Model::InputLogEvent> logs_list_;
+  Aws::SDKOptions options_;
+  std::shared_ptr<CloudWatchFacade> facade_;
+  std::shared_ptr<CloudWatchLogsClientMock> mockClient;
+
+  void SetUp() override
+  {
+    // the tests require non-empty logs_list_
+    logs_list_.emplace_back();
+    logs_list_.emplace_back();
+
+    Aws::InitAPI(options_);
+    mockClient = std::make_shared<CloudWatchLogsClientMock>();
+    facade_ = std::make_shared<CloudWatchFacade>(mockClient);
+  }
+
+  void TearDown() override
+  {
+    Aws::ShutdownAPI(options_);
+  }
+};
+
+/*
+ * SendLogsToCloudWatch Tests
+ */
+
+TEST_F(TestCloudWatchFacade, TestCWLogsFacade_SendLogsToCloudWatch_NullLogs)
+{
+    Aws::String nextToken;
+    EXPECT_EQ(Aws::CloudWatchLogs::ROSCloudWatchLogsErrors::CW_LOGS_NULL_PARAMETER, facade_->SendLogsToCloudWatch(nextToken, "", "", nullptr));
```

Comment:
This and other lines are over the [80 characters limit in the google c++ standard we use](https://google.github.io/styleguide/cppguide.html#Line_Length), so it's difficult to see them in the CR. Can you please make a pass to all the files adding new lines?

```python
>>> len('     EXPECT_EQ(Aws::CloudWatchLogs::ROSCloudWatchLogsErrors::CW_LOGS_SUCCEEDED, facade_->SendLogsToCloudWatch(nextToken, "", "", &logs_list_));')
143
```

Reply:
I think we've decided to follow the ROS2 style guidelines, which 100 characters. This line is still too long by ROS2 standards.